components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+			boolean policyDeleted = pap.deletePolicy(policy);
+			if (policyDeleted) {
+				if (log.isDebugEnabled()) {
+					log.debug("Policy by id:" + policyId + " has been successfully deleted.");
+				}
+				return true;
+			} else {
+				if (log.isDebugEnabled()) {
+					log.debug("Policy by id:" + policyId + " does not exist.");
+				}
+				response.setStatus(Response.Status.NOT_FOUND.getStatusCode());
+				return false;
+			}
+		} catch (PolicyManagementException e) {
+			String error = "Exception in deleting policy by id:" + policyId;
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@PUT
+	@Produces("application/json")
+	@Path("/activate/{id}")
+	public boolean activatePolicy(@PathParam("id") int policyId) {
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.activatePolicy(policyId);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy by id:" + policyId + " has been successfully activated.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Exception in activating policy by id:" + policyId;
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@PUT
+	@Produces("application/json")
+	@Path("/inactivate/{id}")
+	public boolean inactivatePolicy(@PathParam("id") int policyId) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.inactivatePolicy(policyId);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy by id:" + policyId + " has been successfully inactivated.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Exception in inactivating policy by id:" + policyId;
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@PUT
+	@Produces("application/json")
+	@Path("/apply-changes")
+	public boolean applyChanges() {
```

Comment:
why do not we simply return the javax.ws.rs.core.Response object here as well and make the return type consistent across all the APIs?